File: lista de exercicios/questão 2/questao.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self):
        self.velocidade += 10

    def frear(self):
        if self.velocidade >= 10:
            self.velocidade -= 10

    def exibir_velocidade(self):
        print(f"Velocidade atual: {self.velocidade} km/h")

carro = Carro("MarcaA", "ModeloX", 2020)
carro.acelerar()
carro.exibir_velocidade()
carro.frear()
carro.exibir_velocidade()

File: lista de exercicios/questão 2/questao.go
package main

import "fmt"

type Carro struct {
    Marca     string
    Modelo    string
    Ano       int
    Velocidade int
}

func (c *Carro) Acelerar() {
    c.Velocidade += 10
}

func (c *Carro) Frear() {
    if c.Velocidade >= 10 {
        c.Velocidade -= 10
    }
}

func (c Carro) ExibirVelocidade() {
    fmt.Printf("Velocidade atual: %d km/h\n", c.Velocidade)
}

func main() {
    carro := Carro{Marca: "MarcaA", Modelo: "ModeloX", Ano: 2020, Velocidade: 0}
    carro.Acelerar()
    carro.ExibirVelocidade()
    carro.Frear()
    carro.ExibirVelocidade()
}
